### PR TITLE
bug fix karate testing

### PR DESCRIPTION
**Pull request by BelghmiAmine**
This pull request fixes a couple of bugs detected by the karate tests : 

1. Opening the election with open time before election setup results in an error.
2. Casting a valid vote before election opening time should fail.
3. Sending an election end with creation time before election setup should return an error.

### CODE REVIEW

**Review by Ajkunas — CHANGES_REQUESTED** [review]
Reviewed a bit the PR :) I just have some comments concerning some implementations to make it more concise, othw all good

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/ElectionValidator.scala [hunk 1]
```diff
@@ -148,12 +154,23 @@ sealed class ElectionValidator(dbActorRef: => AskableActorRef) extends MessageDa
 
         val electionId = channel.extractChildChannel
 
+        val setupElectionTimeStamp = extractSetupElectionTimestamp(channel, rpcMessage)
+        if (setupElectionTimeStamp.isEmpty) {
```
Comment (Ajkunas) [inline be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/ElectionValidator.scala:158]: Same comment as below. 
I would suggest you to use a match case instead.

**Review by Ajkunas — CHANGES_REQUESTED** [review]
Last nitpicks and it's ready to be approved !

**Review by pierluca — APPROVED** [review]
Approving blindly :)

**Review by Ajkunas — APPROVED** [review]

**Pull request merged**